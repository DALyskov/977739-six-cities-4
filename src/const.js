export const APPROVED_NAME = [
  `Beautiful & luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
];
