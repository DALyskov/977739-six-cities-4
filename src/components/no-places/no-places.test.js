import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

import {ErrReason} from '../../const.js';

import NoPlaces from './no-places.jsx';
import NameSpace from '../../reducer/name-space.js';

const mockStore = configureStore([]);

describe(`NoPlaces_snapchots`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      errMessage: `Error`,
    },
    [NameSpace.STATE_APPLICATION]: {},
    [NameSpace.USER]: {},
  });
  it(`NoPlaces_with_Err`, () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <NoPlaces errReason={ErrReason.LOAD_OFFERS} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it(`NoPlaces_without_Err`, () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <NoPlaces errReason={false} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
