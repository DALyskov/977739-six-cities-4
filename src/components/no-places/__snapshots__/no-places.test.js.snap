// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoPlaces_snapchots NoPlaces_with_Err 1`] = `
<div
  className="cities"
>
  <div
    className="cities__places-container cities__places-container--empty container"
  >
    <section
      className="cities__no-places"
    >
      <div
        className="err-message"
        style={
          Object {
            "color": "red",
            "margin": "10px auto",
            "textAlign": "center",
          }
        }
      >
        Error
      </div>
      <div
        className="cities__status-wrapper tabs__content"
      >
        <b
          className="cities__status"
        >
          No places to stay available
        </b>
        <p
          className="cities__status-description"
        >
          We could not find any property availbale at the moment in Dusseldorf
        </p>
      </div>
    </section>
    <div
      className="cities__right-section"
    />
  </div>
</div>
`;

exports[`NoPlaces_snapchots NoPlaces_without_Err 1`] = `
<div
  className="cities"
>
  <div
    className="cities__places-container cities__places-container--empty container"
  >
    <section
      className="cities__no-places"
    >
      <div
        className="cities__status-wrapper tabs__content"
      >
        <b
          className="cities__status"
        >
          No places to stay available
        </b>
        <p
          className="cities__status-description"
        >
          We could not find any property availbale at the moment in Dusseldorf
        </p>
      </div>
    </section>
    <div
      className="cities__right-section"
    />
  </div>
</div>
`;
